service/raft/raft_group0.cc

```diff
@@ -776,6 +776,10 @@ future<bool> raft_group0::wait_for_raft() {
             "We're fully upgraded to use Raft but didn't join group 0. Please report a bug.");
     }
 
+    group0_log.info("Performing a group 0 read barrier...");
+    co_await _raft_gr.group0().read_barrier(&_abort_source);
```

Comment:
I believe this was explicitly avoided before since we wanted the nodes to restart successfully in absence of a majority. I think we even have dtests for it. @gleb-cloudius

Reply:
Yes. Good catch. I am not sure we have a test, but if we do not we should have one. It is fine to call the barrier during 'removenode' operation though before accessing the config.

Reply:
`wait_for_raft` is not called during restart.

Reply:
checked the code, this is the case indeed.